pkg/zonemgr/zone_file_cache.go

```diff
@@ -53,12 +56,33 @@ func (zoneFileCache *ZoneFileCache) prepare() {
 	zoneFileCache.initCustomFields()
 	zoneFileCache.generateHeaderPrefix()
 	zoneFileCache.generateHeaderSuffix()
-	zoneFileCache.soaSerial = 0
+	zoneFileCache.soaSerial = initSoaSerial(zoneFileNamePrefix + zoneFileCache.domain)
 	zoneFileCache.header = zoneFileCache.generateHeader()
 	zoneFileCache.content = zoneFileCache.header
 	zoneFileCache.vmiRecordsMap = make(map[string][]string)
 }
 
+func initSoaSerial(zoneFileName string) int {
+	if content, err := readFile(zoneFileName); err == nil && content != nil {
+		return fetchSoaSerial(string(content))
+	} else {
+		return 0
+	}
+}
+
+func fetchSoaSerial(content string) int {
+	if result := soaSerialReg.FindStringSubmatch(content); result != nil && len(result) > 0 {
+		soaSerial := result[1]
+		if soaSerialInt, err := strconv.Atoi(soaSerial); err == nil {
+			return soaSerialInt
+		} else {
+			return 0
+		}
+	} else {
+		return 0
+	}
```

Comment:
Lets try please to reduce else and ident
```
serial := 0
if result := soaSerialReg.FindStringSubmatch(content); result != nil && len(result) > 0 {
    soaSerial := result[1]
    if soaSerialInt, err := strconv.Atoi(soaSerial); err == nil {
        serial = soaSerialInt
    } 
}
return serial
```
same can be done for the function above please

Reply:
fixed